src/metar/metar_csv2ioda.py

```diff
@@ -0,0 +1,322 @@
+#!/usr/bin/env python
+
+#
+# (C) Copyright 2020 UCAR
+#
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# author: Greg Thompson gthompsn AT ucar DOT edu
+# This script will work with decoded, CSV files created by gthompsn stored at NCAR
+#   data-access.ucar.edu:/glade/campaign/ral/aap/gthompsn/METARs/2019/20191231/2019123118_metars.csv.gz
+#
+#..Temporarily include the ioda_conv_ncio library path explicitly using this: export PYTHONPATH=../lib-python:$PYTHONPATH
+#
+
+from __future__ import print_function
+import sys
+import math
+from datetime import datetime
+from argparse import ArgumentParser, ArgumentDefaultsHelpFormatter
+from pathlib import Path
+from csv import DictReader
+import netCDF4
+
+IODA_CONV_PATH = Path(__file__).parent/"@SCRIPT_LIB_PATH@"
+if not IODA_CONV_PATH.is_dir():
+    IODA_CONV_PATH = Path(__file__).parent/'..'/'lib-python'
+sys.path.append(str(IODA_CONV_PATH.resolve()))
+
+import meteo_utils
+import ioda_conv_ncio as iconv
+from orddicts import DefaultOrderedDict
+
+class reformatMetar(object):
+
+    def __init__(self, filename, date):
+
+        self.filename = filename
+        self.date = date
+        self.meteo_utils = meteo_utils.meteo_utils()
+
+        # Read in CSV-formatted file of METAR data
+        self._rd_metars()
+
+        return
+
+    def _rd_metars(self):
+
+        # data is the dictionary of incoming observation (METAR) data
+        data = {}
+
+        data['ob_icao'] = []
+        data['ob_lat'] = []
+        data['ob_lon'] = []
+        data['ob_time'] = []
+        data['ob_datetime'] = []
+        data['ob_elev'] = []
+        data['ob_psfc'] = []
+        data['ob_temp'] = []
+        data['ob_spfh'] = []
+        data['ob_uwnd'] = []
+        data['ob_vwnd'] = []
+
+        '''
+        Read in the METARs data
+        Header contains: ICAO, Unix_time, Latitude, Longitude, Elev, MSLP, Altimeter, Temp, Dewp, Wdir, Wspd
+        '''
+
+        # open file in read mode
+        with open(self.filename, 'r') as fh:
+            # pass the file object to reader() to get the reader object
+            csv_dict_reader = DictReader(fh)
+            column_names = csv_dict_reader.fieldnames
+            print(column_names)
```

Comment:
Is this a debug print or is this intended to stay permanently? Shouldn't the column names be constant?

Reply:
It was effectively a debug print.  The column names only need to be constant for the variables that actually get captured (e.g., ICAO) whereas the order could be entirely different since python is capturing by name in header, not by the physical placement/order.  However, if desired, the print statement could be eliminated, although it does seem OK to leave a single print statement per running of the converter, don't you think?

Reply:
Yeah it's probably fine, just checking.